### PR TITLE
plugin/loop: tweak loop detected msg, add troubleshooting section

### PR DESCRIPTION
**Pull request by chrisohaver**
### 1. Why is this pull request needed and what does it do?

Feedback from the field, the current message is confusing some people. This PR tweaks the message a bit, moving the long "confusing" HINFO probe detail to the end of the message.

In the context of kubernetes, the exit on detection behavior manifests as a "crashloop" on the pod, since k8s detects the container exit as a "crash", and will try to start it up again ad infinitum.  I think people also find this confusing. But, there isn't anything practical we could do to change that.  If for example, we didn't exit, and disabled all forwards/proxys to the offending zone (hard to do), it would leave CoreDNS in a "silently" semi-functional state, which is IMO worse.  The presence of the coredns pod "crashloop" is a breadcrumb of sorts, to go check out the coredns log.  So, I think the best we can do is to improve the message a little.

### 2. Which issues (if any) are related?

it has come up in slack many times, and in various github repos... 
e.g. kubernetes/kubeadm#1162

### 3. Which documentation changes (if any) need to be made?
Included

### CODE REVIEW

**Comment by chrisohaver** [comment]
actually - i need to pull the zone part from the probe message - not hard code to "."

File: plugin/loop/loop.go [hunk 1]
```diff
@@ -49,7 +50,11 @@ func (l *Loop) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (
 	}
 
 	if l.seen() > 2 {
-		log.Fatalf("Seen \"HINFO IN %s\" more than twice, loop detected", l.qname)
+		zone := "."
+		if len(dns.SplitDomainName(l.qname)) > 2 {
+			zone = dnsutil.Join(dns.SplitDomainName(l.qname)[2:]...)
```
Comment (miekg) [inline plugin/loop/loop.go]: why is this not `l.zone`?

We should probably also point to https://coredns.io/plugins/loop#looping (and adjust loop/README.md to have a section on it under `#looping`)

Comment (chrisohaver) [reply plugin/loop/loop.go]: ha `l.zone` of course - the answer is already there.

> We should probably also point to

Do you mean put url in the error message? e.g. 

```
Forwarding loop detected in "." zone. See https://coredns.io/plugins/loop#looping. ...
```

**Review by rajansandeep — APPROVED** [review]

File: plugin/loop/loop.go [hunk 2]
```diff
@@ -50,11 +49,7 @@ func (l *Loop) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (
 	}
 
 	if l.seen() > 2 {
-		zone := "."
-		if len(dns.SplitDomainName(l.qname)) > 2 {
-			zone = dnsutil.Join(dns.SplitDomainName(l.qname)[2:]...)
-		}
-		log.Fatalf("Forwarding loop detected in \"%s\" zone. Probe \"HINFO IN %s\" was seen more than twice.", zone, l.qname)
+		log.Fatalf("Forwarding loop detected in \"%s\" zone. Exiting. See https://coredns.io/plugins/loop#troubleshooting.", l.zone, l.qname)
```
Comment (johnbelamaric) [inline plugin/loop/loop.go]: You left l.qname in there, it should come out. Otherwise, looks good.

Comment (johnbelamaric) [reply plugin/loop/loop.go]: ooops, nevermind. was looking at a partial change

**Comment by miekg** [comment]
Yes put the URL there. Means we don't need a new section on coredns.io and
the info is also in the readme and manual page

**Pull request merged**